include/Options/Options.php

```diff
@@ -108,9 +123,13 @@ public function __construct() {
 	 * @phpstan-param class-string<Abstract_Option> $class_name
 	 */
 	public function register( string $class_name ): self {
-		foreach ( $this->options as &$options ) {
-			$key = $class_name::key();
+		$key = $class_name::key();
+
+		if ( ! $this->is_plugin_active() && ! array_key_exists( $key, $this->default ) ) {
+			$this->default[ $key ] = $class_name::get_default();
+		}
 
+		foreach ( $this->options as &$options ) {
 			if ( ! array_key_exists( $key, $options ) ) {
 				// Option raw value doesn't exist in database, use default instead.
 				$options[ $key ] = new $class_name();
```

Comment:
This is what @Chouby wanted to move to `self::get()`.
Along the code you added just before.
This way you won't need to have `Options::$default`.